c/src/sbp.c

```diff
@@ -306,6 +306,12 @@ void sbp_state_set_io_context(sbp_state_t *s, void *context)
  *         has occurred. Thus can check for >0 to ensure good processing.
  */
 s8 sbp_process(sbp_state_t *s, s32 (*read)(u8 *buff, u32 n, void *context))
+{
+return sbp_process_generic(s, read, &sbp_process_payload);
+}
+
+s8 sbp_process_generic(sbp_state_t *s, s32 (*read)(u8 *buff, u32 n, void *context),
+    s8 (*sbp_process_payload) (sbp_state_t *s, u16 sender_id, u16 msg_type, u8 msg_len, u8 payload[]))
```

Comment:
probably want to avoid aliasing `sbp_process_payload` here with the arg name. also you might be able to reuse `sbp_msg_callback_t` if you pass the `sbp_state_t` as the `context *` arg

Reply:
I renamed the existing sbp_process_payload to sbp_process_payload_cbs to notate that it goes through the linked list of callbacks.  This one, I renamed to sbp_process_payload_fn to indicate it is a function pointer.